exonum/src/blockchain/transaction.rs

```diff
@@ -40,7 +39,73 @@ pub type ExecutionResult = Result<(), ExecutionError>;
 #[derive(Clone, Debug, PartialEq)]
 pub struct TransactionResult(pub Result<(), TransactionError>);
 
-/// Transaction processing functionality for `Message`, which allows applying authenticated, atomic,
+/// Data transfer object for transaction.
+/// This structure is used to send api info about transaction,
+/// and take some new transaction into pool from user input.
+#[derive(Serialize, Deserialize)]
+pub struct TransactionMessage {
+    #[serde(skip_deserializing)]
+    #[serde(rename = "debug")]
+    transaction: Option<Box<dyn Transaction>>,
+
+    #[serde(with = "HexStringRepresentation")]
+    message: Message<RawTransaction>,
+}
+impl ::std::fmt::Debug for TransactionMessage {
+    fn fmt(&self, fmt: &mut ::std::fmt::Formatter) -> Result<(), ::std::fmt::Error> {
+        let mut signed_message_debug = String::new();
+        self.message
+            .signed_message()
+            .write_hex(&mut signed_message_debug)?;
```

Comment:
`SignedMessage` may benefit from a `Debug` impl so that we don't have to spell it out explicitly.

Reply:
It has debug impl, but it differ from hexadecimal representation.
I prefer to keep `Debug` info of `TransactionMessage` to be maximum similar with it `Serialized` form.